src/jit/morph.cpp

```diff
@@ -7030,28 +7108,105 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
                 hasMultiByteArgs = true;
             }
         }
+        else
+        {
+            varTypeIsFloating(argx) ? ++calleeFloatArgRegCount : ++calleeArgRegCount;
+        }
     }
 
     // Go the slow route, if it has multi-byte params
     if (hasMultiByteArgs)
     {
+        JITDUMP("Will not fastTailCall hasMultiByteArgs");
+        return false;
+    }
+
+    const unsigned maxRegArgs = MAX_REG_ARG;
+
+// If we reached here means that callee has only those argument types which can be passed in
+// a register and if passed on stack will occupy exactly one stack slot in out-going arg area.
+// If we are passing args on stack for the callee and it has more args passed on stack than
+// the caller, then fast tail call cannot be performed.
+//
+// Note that the GC'ness of on stack args need not match since the arg setup area is marked
+// as non-interruptible for fast tail calls.
+
+#ifdef WINDOWS_AMD64_ABI
+    size_t calleeStackSlots = ((calleeArgRegCount + calleeFloatArgRegCount) > maxRegArgs)
+                                  ? (calleeArgRegCount + calleeFloatArgRegCount) - maxRegArgs
+                                  : 0;
+    size_t calleeStackSize = calleeStackSlots * TARGET_POINTER_SIZE;
+    size_t callerStackSize = info.compArgStackSize;
+
+    // x64 Windows: If we have more callee registers used than MAX_REG_ARG, then
```

Comment:
This comment is confusing. Did you mean "If we have more callee arguments (or slots)"? How can we use more registers than MAX_REG_ARG?